ManageIQ/manageiq
# Linking of Middleware with VMs

Purpose or Intent
-----------------
This PR introduces the `lives_on` to Middleware. Hawkular exposes the `/etc/machine-id` on its `feeds` and we use this value to lookup for an existing managed VM on other providers. If a VM with a matching Machine ID is found, `lives_on` is populated with data from this VM. 

This allows for a later UI navigation between Cloud <-> Hypervisor <-> VM <-> Middleware.

Steps for Testing/QA
--------------------
Set RHEV-M with one Hypervisor and one VM
Add this RHEV-M to MiQ
Install Hawkular Services into the VM
Add this Hawkular instance to MiQ
Check that the `lives_on` for the Hawkular record is populated with data from the VM on the Hypervisor
